doc/how_to/ai_config/custom_data_source.md

```diff
@@ -0,0 +1,42 @@
+# Custom Data Sources
+
+```{admonition} What does this guide solve?
+---
+class: important
+---
+This guide shows you how to configure custom data sources for Lumen AI.
+```
+
+## Overview
+
+We will be using a local LLM to understand how to load custom data sources in Lumen. You do not need
+to use a local LLM, and can instead opt for using one you have an API key for. Lumen currently
+supports the following LLM providers.
+
+- OpenAI
+- Anthropic
+- MistralAI
+- AzureOpenAI
+- AzureMistralAI
+- Llama
+
+Ensure your API key is in the environment of the terminal you run your commands in.
```

Comment:
I know this is early but just as a heads up, each how-to will be pretty standalone. You should assume that a user has configured the LLM already, either by reading the getting started or the how-to on configuring the LLM.